packages/stokhos/src/sacado/kokkos/vector/Stokhos_MP_Vector_MaskTraits.hpp

```diff
@@ -0,0 +1,447 @@
+//
+//                           Stokhos Package
+//                 Copyright (2009) Sandia Corporation
+//
+// Under terms of Contract DE-AC04-94AL85000, there is a non-exclusive
+// license for use of this work by or on behalf of the U.S. Government.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Eric T. Phipps (etphipp@sandia.gov).
+//
+// ***********************************************************************
+// @HEADER
+
+#ifndef STOKHOS_MP_VECTOR_MASKTRAITS_HPP
+#define STOKHOS_MP_VECTOR_MASKTRAITS_HPP
+
+#include "Sacado.hpp"
+#include "Stokhos_Tpetra_Utilities_MP_Vector.hpp"
+#include <iostream>
+
+template <typename T>
+struct EnsembleTraits_m {
+  static const int size = 1;
+  typedef T value_type;
+  static const value_type& coeff(const T& x, int i) { return x; }
+  static value_type& coeff(T& x, int i) { return x; }
+};
+
+template <typename S>
+struct EnsembleTraits_m< Sacado::MP::Vector<S> > {
+  static const int size = S::static_size;
+  typedef typename S::value_type value_type;
+  static const value_type& coeff(const Sacado::MP::Vector<S>& x, int i) {
+    return x.fastAccessCoeff(i);
+  }
+  static value_type& coeff(Sacado::MP::Vector<S>& x, int i) {
+    return x.fastAccessCoeff(i);
+  }
+};
+
+template<typename scalar> class Mask 
+{
+    private: 
+        static const int size = EnsembleTraits_m<scalar>::size;
+        bool data[size];
+
+    public:
+        Mask(){
+        }
+
+        Mask(bool a){
+            size = 1;
+            data = new bool[size]();
```

Comment:
Two issues here.  I'm surprised the compiler doesn't complain about the "size = 1" line, since size is declared const.  Second, please remove the dynamic memory allocation.  I think what you want to do is keep the size the same as defined above using EnsembleTraits_m, and set all entries of data to 1.